### DISCUSSION TITLE
Inventory management in Orca slicer

### DISCUSSION THREAD

**Discussion by hottr** [OrcaSlicer/OrcaSlicer#7262] [Ideas]
Hi, 
Welocme everyone, 
Orca slicer, Bambu studio and Prusa slicer are amazing, Thank you for the amazing pieces of software.

I just opened an account on GitHub to present you my idea for an addition to Orca slicer that would benefit anyone who is tracking filament usage and stock levels, spare parts and consumables. 

This in its own right could be a separate piece of software but integrating it with Orca would simplify things for the user. 

Since Orca can see so much information regarding the printer, filament etc, maybe it would be feasible to add an inventory section to Orca. It could be simple and based on mix of manual and automatic inputs. 

For example:
Filaments: filaments with NFC chip and serial number like BambuLab filaments would be added to library automatically after AMS reads the NFC tag. From this point onward this filament would be in your inventory, and usage would be tracked based on completed prints and information from slicing. 
Same could be done for filaments without the NFC but this would require manual input/recording and selection of the filament loaded in to the printer would also be done manually. maybe it could be combined with QR code stickers for easier tracking. 

Spare parts - manual inputs/records, Allow to create a stock list of most common spare parts for specific printer with stock levels, ability to set minimum stock levels on critical parts, maintenance reminders and similar. 

Additionally it could have a section of "printed objects stocks" if someone is running a printing business to help track stock levels and flows of products. 

project https://github.com/inventree/InvenTree shows very advanced inventory software, maybe this could be done in collaboration with that team?

**Comment by jjayfus** [comment]
And scanning the NFC tag would change the filament profile for your current project. NFC tags are cheep and great for these things.

**Comment by CountGreven** [comment]
I think this is a idea that should be generalized. Define a api for filament inventory systems so orca slicer could load inventory from another program, we could also load filament profiles from that program. This way we keep orca slicer lean and bloat free whilest opening up the door for a comprehensive filament inventory system.

**Comment by sphyx42** [comment]
Wouldn't Spoolman meet these requirements?

**Reply by CountGreven** [reply]
I had never heard of spoolman, thanks :) 
Exactly what I said as well, better to build a integration point for orca to talk with spoolman rather than having orca control the inventory itself :)

**Reply by sphyx42** [reply]
It's kinda funny, I currently run an Inventree server for my school. Then I started up a Spoolman one today and they're actually pretty close! Good luck!

**Reply by Gramps50** [reply]
I have an X1C with AMS & Qidi Plus 4 which runs Kipper. Klipper integrates with Spoolman through Moonraker, you pick the filament to be tracked in Fluid or which ever frontend you use. For non Klipper printers I would like to see an option when the print is sent to the printer to choose which filament(s) are being used from Spoolman. Orca already knows how much filament is going to be used & it could just subtract that amount from Spoolman. This is what I do manually now. Not as accurate as the Klipper way but better than manually doing it. 

This could also work on other printer that aren’t running Klipper & not Bambu. Not perfect but somewhat automated, could be an option that could be on in the printer section for those who wanted to use it.
